apps/dashr-wind-steaming/app.R

```diff
@@ -0,0 +1,443 @@
+
+library(dashR)
+library(dashCoreComponents)
+library(dashHtmlComponents)
+library(purrr)
+library(dplyr)
+library(DBI)
+library(RSQLite)
+library(glue)
+library(data.table)
+library(plotly)
+library(readr)
+library(VGAM)
+
+
+appName <- Sys.getenv("DASH_APP_NAME")
+pathPrefix <- sprintf("/%s/", appName)
+
+Sys.setenv(DASH_ROUTES_PATHNAME_PREFIX = pathPrefix,
+           DASH_REQUESTS_PATHNAME_PREFIX = pathPrefix)
+
+setwd("/app/apps/dashr-opiod-endemic")
+setwd("/app/apps/dashr-wind-streaming")
+
+DB_FILE = "/app/apps/dashr-wind-streaming/db/wind-data.db" 
+
+get_wind_data <- function(start, end) {
+  # Query wind data rows between two ranges
+  # :params start: start row id
+  # :params end: end row id
+  # :returns: pandas dataframe object
+  con <- dbConnect(SQLite(), DB_FILE)
+  res <- dbSendQuery(con, glue("SELECT Speed, SpeedError, Direction FROM Wind WHERE rowid > {start} AND rowid <= {end}"))
+  df <- dbFetch(res)
+  return(df)
+}
+
+get_wind_data_by_id <- function(id){
+  # Query a row from the Wind Table
+  # :params id: a row id
+  # :returns: pandas dataframe object
+  con <- dbConnect(SQLite(), DB_FILE)
+  res <- dbSendQuery(con, glue("SELECT * FROM Wind WHERE rowid = {id}"))
+  df <- dbFetch(res)
+  return(df)
+}
+
+get_current_time <- function(){
+  # Helper function to get the current time in seconds
+  total_time <- as.integer(as.numeric(Sys.time()) / 216000)
+  return(total_time)
+}
+
+app <- Dash$new()
+
+app_color <- list("graph_bg" = "#082255", "graph_line" = "#007ACE")
+
+app$layout(htmlDiv(
+  list(
+    htmlDiv(
+      list(
+        htmlDiv(
+          list(
+            htmlH4("WIND SPEED STREAMING", className = "app__header__title"),
+            htmlP(
+              "This app continually queries a SQL database and displays live charts of wind speed and wind direction.",
+              className = "app__header__title--grey"
+            ),
+            htmlButton(id = "learn-more", src = "https://plot.ly", 
+                       style = list(color = "#082255"))
+          ), className = "app__header__desc"
+        ),
+        htmlDiv(
+          list(
+            htmlImg(
+              src  =  "https://raw.githubusercontent.com/plotly/dash-sample-apps/master/apps/dash-wind-streaming/assets/dash-logo-stripe-inverted.png",
```

Comment:
You can put the latest logo (the `plotly | dash` one) in your assets directory and serve it from there instead